apps/admin_app/lib/admin_app_web/templates/stock_location/_form.html.eex

```diff
@@ -0,0 +1,15 @@
+<%= form_for @changeset, @action, fn f -> %>
+  <%= input f, :name%>
+  <%= input f, :admin_name %>
+  <%= textarea_input f, :address_line_1 %>
+  <%= textarea_input f, :address_line_2 %>
+  <%= input f, :city %>
+  <%= input f, :zip_code %>
+  <%= input f, :phone %>
+  <%= select_input f, :country_id, formated_list(:country) %>
+  <%= select_input f, :state_id, formated_list(:state) %>
```

Comment:
The country needs to be selected first. If the `selected_country.states_required?` is `true`, we require the `state` to be selected.
The relevant changeset validations have been implemented in the [Address schema](https://github.com/aviabird/snitch/blob/develop/apps/snitch_core/lib/core/data/schema/address.ex#L51).
This is because some countries do not have any subdivisions according to ISO 3166-2.

Reply:
This has not been addressed yet. Will you be addressing it in this PR?

Reply:
Nopes not in this.

Reply:
Thanks for letting me know. Please understand that it takes a lot of effort to review large PRs and when a legitimate concern is left out completely unaddressed like this, it quite clearly invalidates the review effort.
This PR is especially hard because there's only one humongous commit. Moreover, the branch is being force pushed again and again (before and even after reviews) which automatically invalidates any effort that has been made in reviewing it.